sdks/go/pkg/beam/core/runtime/xlangx/expansionx/process.go

```diff
@@ -30,20 +31,39 @@ type ExpansionServiceRunner struct {
 	serviceCommand *exec.Cmd
 }
 
+func findOpenPort() (int, error) {
+	listener, err := net.Listen("tcp", ":0")
+	if err != nil {
+		return 0, err
+	}
+	defer listener.Close()
+	return listener.Addr().(*net.TCPAddr).Port, nil
+}
+
 // NewExpansionServiceRunner builds an ExpansionServiceRunner struct for a given gradle target and
 // Beam version and returns a pointer to it.
```

Comment:
Consider adding documentation for the behavior of "Passing an empty string as servicePort will request an open port to be assigned to the servce."

Reply:
Added, totally fair addition